### ISSUE TITLE
Adding most examples from the old repository

### ISSUE THREAD

**Issue by RubenOlsen** [svandecappelle/jQPlot#47]
#  Description

There is a lot of good examples in the old repo. Most of these should be present on the new Github pages.
# Todo

Convert by a simple perl or awk script the old examples into Jekyll-form

# Done criteria

When all the old examples is available on the GH pages

**Comment by svandecappelle** [comment]
Might be cool to add the old examples but it really could be hard to migrate old scripts into the new site demo style. 
Question: how to you think my new version of the demos page ? (using screenshots on main page & codemirror as code viewer)

If migration script is too complexe say me and I will migrate by hand all examples this week.

**Comment by RubenOlsen** [comment]
Hi,

It is extremely easy to migrate all the old scripts. I think I will have a success rate of 90-90% using a few lines of perl. I have for more than 20 years worked a lot on converting documents and data from one format to another format - so this is really a walk in the park for me.

Btw, I have moved away from codemirror and back to syntaxhighligther as this does require less scaffolding than codemirror. Codemirror is nice, but a bit over the top for our use.

I have pushed by version 0.0.2 onto my GH-pages - see http://rubenolsen.github.io/jQPlot/demo/core/Basic%20graph.html <http://rubenolsen.github.io/jQPlot/demo/core/Basic%20graph.html> as a Jekyll'ed version of your first example.  If you need to take a look at the Jekyll sources without me issuing a pull request - go to https://github.com/RubenOlsen/jQPlot/tree/gh-pages

The only stuff that is actually present in the Basic graph.html file is this:

---
layout: demoexamples
title: Basic line graph
---

<h2 class="plain">The most basic line graph</h2>

<p>The most basic jqPlot chart takes a series of data and plots a line.  No options need to be supplied.
	Data is passed in as an array of series.  A series can be either an array of y values or an array of [x,y] data pairs.
	If y values only, x values are assigned like 1, 2, 3, ...  Note, for this plot you don't need any plugins.</p>

<script class="code" type="text/javascript">
	$(document).ready(function(){
		var plot1 = $.jqplot ('chart1', [[3,7,9,1,4,6,8,2,5]]);
	});
</script>


The rest is Jekyll taking care of.

For all practical purposes, this means that in the old examples my script will keep everything between the following two entities in each of the old example files:


....


I have only run into a snag with Github Pages where they do not allow 3rd party Jekyll plugins. I use a plugin now to automatically generate the left hand menu in the demo-examples. However, this is easy-peasy to fix and I'll expect by the end of today that I have converted the current GH pages site into Jekyll.  By tomorrow evening, I'll probably also have 90-90% of all the old examples in place. 

I really like the idea of having screenshots on the main demo page - and will continue to do this - or a variant there of. The only thing that might be an issue is the number of screenshots eventually. When we get a lot of examples in place we'll have to consider grouping the into what kind of graph people want. 

Seen from a user's perspective - a user does not care if they will use a jQPlot plugin or not. End users need a line-diagram, bar-chart, or whatever - and will choose the example based on her requirement. We'll decide on this organization when we get there. 

I'ts better that you use your time on fixing bugs in the code - rather than spending time on the documentation. I have said that I'll take care of the documentation - and I'm fully invested into doing just this (meaning spending 1-2 hours a day until we reach a decent enough version :-)

\Ruben

**Comment by svandecappelle** [comment]
For information: RadarRenderer chart is not yet finished (this is a new feature i work on for jqplot) (that's why i didn't yet added to demos page) For the others it looks very pretty and simple to use.

The navigation on left side is a good idea.

There seems to be an issue on your example:
Some CSS styles doesn't work well.

Example with the donut renderer:
The option: showDataCategoryOnHover : true doesn't work well
Take a look at: http://svandecappelle.github.io/jQPlot/demo/plugins/donut/donut.html

All tooltips are broken in your demo instance. (displayed above visible page part)

**Comment by RubenOlsen** [comment]
Yes I am aware of these tooltips things - this is the next thing on my plate after converting the old examples into jekyll.

I have just barely started looking into the CSS. I have noticed a few others which may or may not related to what you have seen. Since we for all practical purposes are using several CSS files for renderig the demos with some overlaps, there will be issues. However, I have a plan on how to attach these issues. I'm going to create an task in the issue tracker for this.

Regarding the radar chart - I have an simple to implement idea on how to enable/disable examples so that they do not turn up in the listing on the left hand side. We might need this for other demos later on.

\R.

On 10. aug. 2015, at 09:50, Vandecappelle <notifications@github.com> wrote:

> For information: RadarRenderer chart is not yet finished (this is a new feature i work on for jqplot) (that's why i didn't yet added to demos page) For the others it looks very pretty and simple to use.
> 
> The navigation on left side is a good idea.
> 
> There seems to be an issue on your example:
> Some CSS styles doesn't work well.
> 
> Example with the donut renderer:
> The option: showDataCategoryOnHover : true doesn't work well
> Take a look at: http://svandecappelle.github.io/jQPlot/demo/plugins/donut/donut.html
> 
> All tooltips are broken in your demo instance. (displayed above visible page part)
> 
> —
> Reply to this email directly or view it on GitHub.
> 

---  
Med vennlig hilsen / Best regards,                                   Skype:   RubenOlsen
Ove Ruben R Olsen, MSc IT               Cell phone +47 91 57 87 48   Twitter: @RubenOlsen
Chief Technical Officer, Azuralis AS    Desk phone +47 55 62 18 08   Jabber:  ruben@azuralis.no
